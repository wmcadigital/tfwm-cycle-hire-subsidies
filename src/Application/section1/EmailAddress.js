import { useFormState } from "react-final-form";

import FormSection from "../../common/FormSection";
import ProgressIndicator from "../../common/ProgressIndicator";
import Question from "../../common/Question";
import TextInput from "../../common/TextInput";
import { required, email, composeValidators } from "../../common/validation";

const EmailAddress = () => {
  const stateApi = useFormState();

  const formValues = stateApi.values;
  const question = "What is your email address?";
  formValues["registerForYourself"] === "yes";

  const errorEmailAddress =
    stateApi.submitFailed && stateApi.hasValidationErrors
      ? stateApi.errors?.formData.EmailAddress
      : null;

  return (
    <FormSection>
      <ProgressIndicator
        sectionPosition="Section 1 of 2"
        sectionName="About you"
      />
      <Question text={question} />
      <p>Weâ€™ll only use this to contact you about your application.</p>
      <TextInput
        fieldName="formData.EmailAddress"
        label={
          <>
            Email address
            <p>For example, name@example.com</p>
          </>
        }
        error={errorEmailAddress}
        validation={composeValidators(required, email)}
      />
    </FormSection>
  );
};

export default EmailAddress;
