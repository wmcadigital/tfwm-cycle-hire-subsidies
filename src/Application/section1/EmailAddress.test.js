import { useFormState } from "react-final-form";
import { createRenderer } from "react-test-renderer/shallow";

import EmailAddress from "./EmailAddress";

jest.mock("react-final-form");

describe("EmailAddress", () => {
  it("individual is filling out form for themselves", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        registerForYourself: "yes",
      },
    }));
    const renderer = createRenderer();

    renderer.render(<EmailAddress />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("individual is filling out form for someone else", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        registerForYourself: "no",
        firstName: "Test",
        lastName: "Testerson",
      },
    }));
    const renderer = createRenderer();

    renderer.render(<EmailAddress />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("renders with an error", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        registerForYourself: "yes",
      },
      submitFailed: true,
      errors: { emailAddress: "Required" },
    }));
    const renderer = createRenderer();

    renderer.render(<EmailAddress />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
