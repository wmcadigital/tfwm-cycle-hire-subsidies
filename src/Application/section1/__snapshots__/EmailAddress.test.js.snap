// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailAddress individual is filling out form for someone else 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="About you"
    sectionPosition="Section 1 of 2"
  />
  <Question
    text="What is your email address?"
  />
  <p>
    We’ll only use this to contact you about your application.
  </p>
  <TextInput
    error={null}
    fieldName="formData.EmailAddress"
    label={
      <React.Fragment>
        Email address
        <p>
          For example, name@example.com
        </p>
      </React.Fragment>
    }
    validation={[Function]}
  />
</FormSection>
`;

exports[`EmailAddress individual is filling out form for themselves 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="About you"
    sectionPosition="Section 1 of 2"
  />
  <Question
    text="What is your email address?"
  />
  <p>
    We’ll only use this to contact you about your application.
  </p>
  <TextInput
    error={null}
    fieldName="formData.EmailAddress"
    label={
      <React.Fragment>
        Email address
        <p>
          For example, name@example.com
        </p>
      </React.Fragment>
    }
    validation={[Function]}
  />
</FormSection>
`;

exports[`EmailAddress renders with an error 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="About you"
    sectionPosition="Section 1 of 2"
  />
  <Question
    text="What is your email address?"
  />
  <p>
    We’ll only use this to contact you about your application.
  </p>
  <TextInput
    fieldName="formData.EmailAddress"
    label={
      <React.Fragment>
        Email address
        <p>
          For example, name@example.com
        </p>
      </React.Fragment>
    }
    validation={[Function]}
  />
</FormSection>
`;
